src/AppBundle/Listener/ResponseListener.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace AppBundle\Listener;
+
+use AppBundle\Security\RateLimiter;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+
+class ResponseListener
+{
+    private $rateLimiter;
+    private $env;
+
+    public function __construct(RateLimiter $rateLimiter, string $env)
+    {
+        $this->rateLimiter = $rateLimiter;
+        $this->env = $env;
+    }
+
+    public function onKernelResponse(FilterResponseEvent $event)
+    {
+        if (!$event->isMasterRequest() || $this->env === 'test') {
```

Comment:
Is this actually required? We may include the headers even in test environment. It also guess should be added even if we are not responding to the master request.